### ISSUE TITLE
multiple nodes crashed during multi-nemesis (3 parallel nemesis)

### ISSUE THREAD

**Issue by fgelcer** [scylladb/scylla#5765]
5 db nodes
4 loaders

scylla 3.3 - ami-0e13d9a7078e22943
the scenario is longevity-2tb-7days-1Dis-2NonDis-Nemesis (but it actually runs 4 days, not 7)

we run in this scenario 3 parallel nemesis:
* major compaction
* network reject inter-node communication
* stop_wait_start_scylla

longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-6:
```
(CoreDumpEvent Severity.CRITICAL): node=Node longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-6 [34.249.121.136 | 10.0.130.74] (seed: False)
corefile_url=
https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.26d0166efcd64880b000807887ea174e.5490.1581204965000000/core.scylla.996.26d0166efcd64880b000807887ea174e.5490.1581204965000000.gz
backtrace=           PID: 5490 (scylla)
           UID: 996 (scylla)
           GID: 1001 (scylla)
        Signal: 6 (ABRT)
     Timestamp: Sat 2020-02-08 23:36:05 UTC (2min 58s ago)
  Command Line: /usr/bin/scylla --blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 0-11
    Executable: /opt/scylladb/libexec/scylla
 Control Group: /scylla.slice/scylla-server.slice/scylla-server.service
          Unit: scylla-server.service
         Slice: scylla-server.slice
       Boot ID: 26d0166efcd64880b000807887ea174e
    Machine ID: df877a200226bc47d06f26dae0736ec9
      Hostname: longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-6
      Coredump: /var/lib/systemd/coredump/core.scylla.996.26d0166efcd64880b000807887ea174e.5490.1581204965000000
       Message: Process 5490 (scylla) of user 996 dumped core.
                
                Stack trace of thread 5490:
                #0  0x00007f9862f38625 raise (libc.so.6)
                #1  0x00007f9862f219ae abort (libc.so.6)
                #2  0x00007f9862f217a9 __assert_fail_base.cold (libc.so.6)
                #3  0x00007f9862f30a66 __assert_fail (libc.so.6)
                #4  0x0000000001ecb369 _ZN7seastar7shardedIN7service17migration_managerEE5localEv (scylla)
                #5  0x00000000019e9da4 operator() > (scylla)
                #6  0x00000000019ea48d operator() (scylla)
                #7  0x00000000007729ed _ZNK7seastar20noncopyable_functionIFvvEEclEv (scylla)
                
                Stack trace of thread 5506:
                #0  0x00007f986396b87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98639614e2 start_thread (libpthread.so.0)
                #5  0x00007f9862ffd693 __clone (libc.so.6)
                
                Stack trace of thread 5512:
                #0  0x00007f986396b87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98639614e2 start_thread (libpthread.so.0)
                #5  0x00007f9862ffd693 __clone (libc.so.6)
                
                Stack trace of thread 5504:
                #0  0x00007f986396b87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98639614e2 start_thread (libpthread.so.0)
                #5  0x00007f9862ffd693 __clone (libc.so.6)
                
                Stack trace of thread 5509:
                #0  0x00007f986396b87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98639614e2 start_thread (libpthread.so.0)
                #5  0x00007f9862ffd693 __clone (libc.so.6)
                
                Stack trace of thread 5513:
                #0  0x00007f986396b87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98639614e2 start_thread (libpthread.so.0)
                #5  0x00007f9862ffd693 __clone (libc.so.6)
                
                Stack trace of thread 5511:
                #0  0x00007f986396b87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98639614e2 start_thread (libpthread.so.0)
                #5  0x00007f9862ffd693 __clone (libc.so.6)
                
                Stack trace of thread 5503:
                #0  0x00007f986396b87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98639614e2 start_thread (libpthread.so.0)
                #5  0x00007f9862ffd693 __clone (libc.so.6)
                
                Stack trace of thread 5510:
                #0  0x00007f986396b87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98639614e2 start_thread (libpthread.so.0)
                #5  0x00007f9862ffd693 __clone (libc.so.6)
                
                Stack trace of thread 5508:
                #0  0x00007f986396b87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98639614e2 start_thread (libpthread.so.0)
                #5  0x00007f9862ffd693 __clone (libc.so.6)
                
                Stack trace of thread 5493:
                #0  0x00007f9862ff81ad syscall (libc.so.6)
                #1  0x0000000002de7eda _ZN7seastar8internal13io_pgeteventsEmllPNS0_9linux_abi8io_eventEPK8timespecPK10__sigset_tb (scylla

download_instructions=
gsutil cp gs://upload.scylladb.com/core.scylla.996.26d0166efcd64880b000807887ea174e.5490.1581204965000000/core.scylla.996.26d0166efcd64880b000807887ea174e.5490.1581204965000000.gz .
gunzip /var/lib/systemd/coredump/core.scylla.996.26d0166efcd64880b000807887ea174e.5490.1581204965000000.gz	

```

longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-8:
```
2020-02-08 23:36:06.000	(CoreDumpEvent Severity.CRITICAL): node=Node longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-8 [34.254.188.65 | 10.0.72.170] (seed: False)
corefile_url=
https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.5dee9e60a5354280a1b019effe7e671e.5478.1581204966000000/core.scylla.996.5dee9e60a5354280a1b019effe7e671e.5478.1581204966000000.gz
backtrace=           PID: 5478 (scylla)
           UID: 996 (scylla)
           GID: 1001 (scylla)
        Signal: 6 (ABRT)
     Timestamp: Sat 2020-02-08 23:36:06 UTC (2min 56s ago)
  Command Line: /usr/bin/scylla --blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 0-11
    Executable: /opt/scylladb/libexec/scylla
 Control Group: /scylla.slice/scylla-server.slice/scylla-server.service
          Unit: scylla-server.service
         Slice: scylla-server.slice
       Boot ID: 5dee9e60a5354280a1b019effe7e671e
    Machine ID: df877a200226bc47d06f26dae0736ec9
      Hostname: longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-8
      Coredump: /var/lib/systemd/coredump/core.scylla.996.5dee9e60a5354280a1b019effe7e671e.5478.1581204966000000
       Message: Process 5478 (scylla) of user 996 dumped core.
                
                Stack trace of thread 5478:
                #0  0x00007fee21fba625 raise (libc.so.6)
                #1  0x00007fee21fa39ae abort (libc.so.6)
                #2  0x00007fee21fa37a9 __assert_fail_base.cold (libc.so.6)
                #3  0x00007fee21fb2a66 __assert_fail (libc.so.6)
                #4  0x0000000001ecb369 _ZN7seastar7shardedIN7service17migration_managerEE5localEv (scylla)
                #5  0x00000000019e9da4 operator() > (scylla)
                #6  0x00000000019ea48d operator() (scylla)
                #7  0x00000000007729ed _ZNK7seastar20noncopyable_functionIFvvEEclEv (scylla)
                
                Stack trace of thread 5492:
                #0  0x00007fee229ed87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007fee229e34e2 start_thread (libpthread.so.0)
                #5  0x00007fee2207f693 __clone (libc.so.6)
                
                Stack trace of thread 5495:
                #0  0x00007fee229ed87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007fee229e34e2 start_thread (libpthread.so.0)
                #5  0x00007fee2207f693 __clone (libc.so.6)
                
                Stack trace of thread 5493:
                #0  0x00007fee229ed87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007fee229e34e2 start_thread (libpthread.so.0)
                #5  0x00007fee2207f693 __clone (libc.so.6)
                
                Stack trace of thread 5501:
                #0  0x00007fee229ed87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007fee229e34e2 start_thread (libpthread.so.0)
                #5  0x00007fee2207f693 __clone (libc.so.6)
                
                Stack trace of thread 5500:
                #0  0x00007fee229ed87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007fee229e34e2 start_thread (libpthread.so.0)
                #5  0x00007fee2207f693 __clone (libc.so.6)
                
                Stack trace of thread 5490:
                #0  0x00007fee229ed87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007fee229e34e2 start_thread (libpthread.so.0)
                #5  0x00007fee2207f693 __clone (libc.so.6)
                
                Stack trace of thread 5491:
                #0  0x00007fee229ed87c read (libpthread.so.0)
                #1  0x0000000002de57d6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002de5ac2 operator() (scylla)
                #3  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007fee229e34e2 start_thread (libpthread.so.0)
                #5  0x00007fee2207f693 __clone (libc.so.6)
                
                Stack trace of thread 5489:
                #0  0x00007fee2207a1ad syscall (libc.so.6)
                #1  0x0000000002de7eda _ZN7seastar8internal13io_pgeteventsEmllPNS0_9linux_abi8io_eventEPK8timespecPK10__sigset_tb (scylla)
                #2  0x0000000002de47f9 _ZN7seastar19reactor_backend_aio12await_eventsEiPK10__sigset_t (scylla)
                #3  0x0000000002de4bbd _ZN7seastar19reactor_backend_aio16wait_and_processEiPK10__sigset_t (scylla)
                #4  0x00000000029f4bf5 _ZN7seastar7reactor16wait_and_processEiPK10__sigset_t (scylla)
                #5  0x0000000002a35278 _ZN7seastar7reactor3runEv (scylla)
                #6  0x0000000002a43a3d _ZZN7seastar3smp9configureEN5boost15program_options13variables_mapENS_14reactor_configEENKUlvE1_clEv (scylla)
                #7  0x00000000029cecae _ZNKSt8functionIFvvEEclEv (scylla)
                #8  0x00007fee229e34e2 start_thread (libpthread.so.0)
                #9  0x00007fee2207f693 __clone (libc.so.6)
                
                Stack trace of thread 5481:
                #0  0x00007fee2207a1ad syscall (libc.so.6)
                #1  0x0000000002de7eda _ZN7seastar8i

download_instructions=
gsutil cp gs://upload.scylladb.com/core.scylla.996.5dee9e60a5354280a1b019effe7e671e.5478.1581204966000000/core.scylla.996.5dee9e60a5354280a1b019effe7e671e.5478.1581204966000000.gz .
gunzip /var/lib/systemd/coredump/core.scylla.996.5dee9e60a5354280a1b019effe7e671e.5478.1581204966000000.gz
```

decoded backtrace from node longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-8:
```
2020-02-08 23:36:06.000: (DatabaseLogEvent Severity.ERROR): type=BACKTRACE regex=backtrace line_number=76514 node=Node longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-8 [34.254.188.65 | 10.0.72.170] (seed: False)
2020-02-08T23:36:06+00:00  longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-8 !INFO    | scylla: Backtrace:

void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/include/fmt/format.h:1158
seastar::backtrace_buffer::append_backtrace() at /usr/include/fmt/format.h:1158
 (inlined by) print_with_backtrace at ./build/release/seastar/./seastar/src/core/reactor.cc:742
seastar::print_with_backtrace(char const*) at /usr/include/fmt/format.h:1158
sigabrt_action at /usr/include/fmt/format.h:1158
 (inlined by) operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3406
 (inlined by) _FUN at ./build/release/seastar/./seastar/src/core/reactor.cc:3402
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
seastar::sharded<service::migration_manager>::local() at /usr/include/fmt/format.h:1158
 (inlined by) service::get_local_migration_manager() at ././service/migration_manager.hh:165
 (inlined by) service::storage_service::on_alive(gms::inet_address, gms::endpoint_state) at ./service/storage_service.cc:1203
operator()<seastar::shared_ptr<gms::i_endpoint_state_change_subscriber> > at /usr/include/fmt/format.h:1158
 (inlined by) for_each<std::_List_iterator<seastar::shared_ptr<gms::i_endpoint_state_change_subscriber> >, gms::gossiper::real_mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda(auto:186&)> > at /usr/include/c++/9/bits/stl_algo.h:3876
 (inlined by) for_each<gms::gossiper::real_mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda(auto:186&)> > at ././gms/gossiper.hh:219
 (inlined by) gms::gossiper::real_mark_alive(gms::inet_address, gms::endpoint_state&) at ./gms/gossiper.cc:1468
operator() at /usr/include/fmt/format.h:1158
 (inlined by) apply at ././seastar/include/seastar/core/apply.hh:36
 (inlined by) apply<gms::gossiper::mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda()>::<lambda()> > at ././seastar/include/seastar/core/apply.hh:44
 (inlined by) apply_tuple<gms::gossiper::mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda()>::<lambda()> > at ././seastar/include/seastar/core/future.hh:1599
 (inlined by) apply<gms::gossiper::mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda()>::<lambda()> > at ././seastar/include/seastar/core/future.hh:1633
 (inlined by) operator() at ././seastar/include/seastar/core/thread.hh:265
 (inlined by) call at ././seastar/include/seastar/util/noncopyable_function.hh:101
seastar::noncopyable_function<void ()>::operator()() const at ./build/release/seastar/./seastar/include/seastar/util/noncopyable_function.hh:184
 (inlined by) seastar::thread_context::main() at ./build/release/seastar/./seastar/src/core/thread.cc:277
```

although we didn't catch the coredump for node longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-7, there is the decoded backtrace (that was caught afterwards):
```
2020-02-08 23:36:06.000: (DatabaseLogEvent Severity.ERROR): type=BACKTRACE regex=backtrace line_number=89353 node=Node longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-7 [34.255.208.107 | 10.0.6.201] (seed: False)
2020-02-08T23:36:06+00:00  longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-7 !INFO    | scylla: Backtrace:

void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/include/fmt/format.h:1158
seastar::backtrace_buffer::append_backtrace() at /usr/include/fmt/format.h:1158
 (inlined by) print_with_backtrace at ./build/release/seastar/./seastar/src/core/reactor.cc:742
seastar::print_with_backtrace(char const*) at /usr/include/fmt/format.h:1158
sigabrt_action at /usr/include/fmt/format.h:1158
 (inlined by) operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3406
 (inlined by) _FUN at ./build/release/seastar/./seastar/src/core/reactor.cc:3402
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
seastar::sharded<service::migration_manager>::local() at /usr/include/fmt/format.h:1158
 (inlined by) service::get_local_migration_manager() at ././service/migration_manager.hh:165
 (inlined by) service::storage_service::on_alive(gms::inet_address, gms::endpoint_state) at ./service/storage_service.cc:1203
operator()<seastar::shared_ptr<gms::i_endpoint_state_change_subscriber> > at /usr/include/fmt/format.h:1158
 (inlined by) for_each<std::_List_iterator<seastar::shared_ptr<gms::i_endpoint_state_change_subscriber> >, gms::gossiper::real_mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda(auto:186&)> > at /usr/include/c++/9/bits/stl_algo.h:3876
 (inlined by) for_each<gms::gossiper::real_mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda(auto:186&)> > at ././gms/gossiper.hh:219
 (inlined by) gms::gossiper::real_mark_alive(gms::inet_address, gms::endpoint_state&) at ./gms/gossiper.cc:1468
operator() at /usr/include/fmt/format.h:1158
 (inlined by) apply at ././seastar/include/seastar/core/apply.hh:36
 (inlined by) apply<gms::gossiper::mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda()>::<lambda()> > at ././seastar/include/seastar/core/apply.hh:44
 (inlined by) apply_tuple<gms::gossiper::mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda()>::<lambda()> > at ././seastar/include/seastar/core/future.hh:1599
 (inlined by) apply<gms::gossiper::mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda()>::<lambda()> > at ././seastar/include/seastar/core/future.hh:1633
 (inlined by) operator() at ././seastar/include/seastar/core/thread.hh:265
 (inlined by) call at ././seastar/include/seastar/util/noncopyable_function.hh:101
seastar::noncopyable_function<void ()>::operator()() const at ./build/release/seastar/./seastar/include/seastar/util/noncopyable_function.hh:184
 (inlined by) seastar::thread_context::main() at ./build/release/seastar/./seastar/src/core/thread.cc:277
```

decoded backtrace from node longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-6:
```
2020-02-08 23:36:05.000: (DatabaseLogEvent Severity.ERROR): type=BACKTRACE regex=backtrace line_number=115421 node=Node longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-6 [34.249.121.136 | 10.0.130.74] (seed: False)
2020-02-08T23:36:05+00:00  longevity-tls-2tb-4d-1dis-2nondis-3-db-node-1dcb436d-6 !INFO    | scylla: Backtrace:

void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/include/fmt/format.h:1158
seastar::backtrace_buffer::append_backtrace() at /usr/include/fmt/format.h:1158
 (inlined by) print_with_backtrace at ./build/release/seastar/./seastar/src/core/reactor.cc:742
seastar::print_with_backtrace(char const*) at /usr/include/fmt/format.h:1158
sigabrt_action at /usr/include/fmt/format.h:1158
 (inlined by) operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3406
 (inlined by) _FUN at ./build/release/seastar/./seastar/src/core/reactor.cc:3402
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
seastar::sharded<service::migration_manager>::local() at /usr/include/fmt/format.h:1158
 (inlined by) service::get_local_migration_manager() at ././service/migration_manager.hh:165
 (inlined by) service::storage_service::on_alive(gms::inet_address, gms::endpoint_state) at ./service/storage_service.cc:1203
operator()<seastar::shared_ptr<gms::i_endpoint_state_change_subscriber> > at /usr/include/fmt/format.h:1158
 (inlined by) for_each<std::_List_iterator<seastar::shared_ptr<gms::i_endpoint_state_change_subscriber> >, gms::gossiper::real_mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda(auto:186&)> > at /usr/include/c++/9/bits/stl_algo.h:3876
 (inlined by) for_each<gms::gossiper::real_mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda(auto:186&)> > at ././gms/gossiper.hh:219
 (inlined by) gms::gossiper::real_mark_alive(gms::inet_address, gms::endpoint_state&) at ./gms/gossiper.cc:1468
operator() at /usr/include/fmt/format.h:1158
 (inlined by) apply at ././seastar/include/seastar/core/apply.hh:36
 (inlined by) apply<gms::gossiper::mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda()>::<lambda()> > at ././seastar/include/seastar/core/apply.hh:44
 (inlined by) apply_tuple<gms::gossiper::mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda()>::<lambda()> > at ././seastar/include/seastar/core/future.hh:1599
 (inlined by) apply<gms::gossiper::mark_alive(gms::inet_address, gms::endpoint_state&)::<lambda()>::<lambda()> > at ././seastar/include/seastar/core/future.hh:1633
 (inlined by) operator() at ././seastar/include/seastar/core/thread.hh:265
 (inlined by) call at ././seastar/include/seastar/util/noncopyable_function.hh:101
seastar::noncopyable_function<void ()>::operator()() const at ./build/release/seastar/./seastar/include/seastar/util/noncopyable_function.hh:184
 (inlined by) seastar::thread_context::main() at ./build/release/seastar/./seastar/src/core/thread.cc:277
```

**Comment by slivne** [comment]
@xemul is this something you had changes in recently ?

**Comment by xemul** [comment]
The very very similar issue must have been fixed with 4839ca849136841b4545931876b1345ee09affa1 , but it still was not the end of the story

**Comment by slivne** [comment]
@xemul can you please look on 3.3 branch specifically related to this case and see if it makes sense to backport the commit you provided - if there are followup patches they may be considered as well - yet we do not want to backport everything

**Comment by xemul** [comment]
@slivne , the patch 4839ca849136841b4545931876b1345ee09affa1 from master is surely needed. 

There's also a patchset titled "[PATCH 0/3] Fix use-after-frees in migration_manager and feature_service" on the mailing list awaiting for @tgrabiec review. This set fixes other problems with migration manager (and database), none of them was captured explicitly (as far as I know). If they hit the master I would also backport patches 1 and 2 of it.

**Comment by slivne** [comment]
@xemul can you please clarify which branches are vulnerable to this 3.3 only or others ?

**Comment by xemul** [comment]
@slivne , 3.* are all affected. Which of 2.*, 1.* and 0.* should I check?

**Comment by bhalevy** [comment]
> @slivne , 3.* are all affected. Which of 2.* , 1.* and 0.* should I check?

@xemul we backport now only 3.1 and newer release, and to enterprise 2019.1

**Comment by slivne** [comment]
@avikivity ping on xemul patch above

**Comment by tgrabiec** [comment]
The patch was merged yesterday, but then reverted. I merged v2 just now.

**Comment by slivne** [comment]
thanks @tgrabiec

**Comment by tgrabiec** [comment]
@xemul We didn't stop migration_manager in 3.1 (we do since f6ac969f1eb5ef504ea569a449b78ff7c730326b), so I think we are not vulnerable beyond master. Am I missing something?

**Comment by xemul** [comment]
@tgrabiec , you're right. I was thinking about aborted start case, but storage service subscribes on gossiper after migration manager starts, so we're safe here too.

**Comment by slivne** [comment]
@tgrabiec can you please do the backport if the code was merged - we failed in 3.3. testing on this.

**Comment by tgrabiec** [comment]
@xemul Can you please prepare a branch on top of 3.3, it doesn't apply cleanly.

**Comment by xemul** [comment]
@tgrabiec https://github.com/xemul/scylla/tree/br-mm-combined-fixes-for-3.3
It includes efd7efe4, 08363e50 and 4839ca84
Tested with unit(dev)

**Comment by slivne** [comment]
@tgrabiec can you please review + merge

**Comment by slivne** [comment]
merged as scylladb/scylla@7b2eb092255f71d4e51dc7f4e5a4754a210e494d
